pkg/storageincentives/agent.go

```diff
@@ -302,6 +311,61 @@ func (a *Agent) start(blockTime time.Duration, blocksPerRound, blocksPerPhase ui
 	}
 }
 
+func (a *Agent) startOutdatedDataPurger(onNewRoundC <-chan uint64) {
```

Comment:
If we do this on every round, then we will mostly be deleting 1 entry per round. Having a separate goroutine for deleting single entry seems to be too much. Instead we can trigger this on the end of the Claim phase and not have this routine.